fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineScrubber.java

```diff
@@ -0,0 +1,918 @@
+         *
+         * <pre><code>
+         * OnlineScrubber.ScrubbingPolicy.newBuilder().setLogWarningsLimit(100).setAllowRepair(true).build()
+         * </code></pre>
+         *
+         */
+        @API(API.Status.UNSTABLE)
+        public static class Builder {
+            int logWarningsLimit = 1000;
+            boolean allowRepair = true;
+            long entriesScanLimit = 0;
+
+            protected Builder() {
+            }
+
+            /**
+             * Set a rigid limit on the max number of warnings to log..
+             * If never called, the default is allowing (up to) 1000 warnings.
+             * @param logWarningsLimit the max number of warnings to log.
+             * @return this builder.
+             */
+            public Builder setLogWarningsLimit(final int logWarningsLimit) {
+                this.logWarningsLimit = logWarningsLimit;
+                return this;
+            }
+
+
+            /**
+             * Set whether the scrubber, if it finds an error, will repair it.
+             * @param val - if false, always report errors but do not repair.
+             *            - if true (default), report errors up to requested limits, but always repair.
+             * @return this builder.
+             */
+            public Builder setAllowRepair(boolean val) {
+                this.allowRepair = val;
+                return this;
+            }
+
+            /**
+             * Set records/index entries scan limit.
+             * Note that for efficiency, the scrubber reads and processes chunks of data, and will only check this limit
+             * after processing a chunk. The scrubber will either stop when the number of checked entries exceeds this limit
+             * or when it covers the whole range.
+             * If stopped by limit, the next scrubber call will skipped the already checked ranges. If the previous
+             * scrubber calls had exhausted the whole range, it'll start a fresh scan.
+             * @param entriesScanLimit - if 0 (default) or less, unlimited. Else return after scanned records count exceeds this limit.
+             * @return this builder.
+             */
+            public Builder setEntriesScanLimit(long entriesScanLimit) {
+                this.entriesScanLimit = entriesScanLimit;
+                return this;
+            }
+
+            public ScrubbingPolicy build() {
+                return new ScrubbingPolicy(logWarningsLimit, allowRepair, entriesScanLimit);
+            }
+        }
+    }
+
+    /**
+     * Builder for {@link OnlineScrubber}.
+     *
+     * <pre><code>
+     * OnlineScrubber.newBuilder().setRecordStoreBuilder(recordStoreBuilder).setIndex(index).build()
+     * </code></pre>
+     *
+     * <pre><code>
+     * OnlineScrubber.newBuilder().setDatabase(fdb).setMetaData(metaData).setSubspace(subspace).setIndex(index).build()
+     * </code></pre>
+     *
+     */
+    @API(API.Status.UNSTABLE)
+    public static class Builder {
```

Comment:
You mentioned in a conversation not on GitHub that you didn't like the amount of code duplication between this and the `OnlineIndexer.Builder`. I agree, and I _think_ that the most straightforward way to remove code dupe might be to have something like a "abstract index processor builder" or something that contains the configuration parameters relevant to both this and indexing and then handles creating an `OnlineIndexer.Config`, etc. I think it can be done in a way that's at least API (if not ABI) backwards compatible. But also not something that I think needs to be done before the scrubbing logic can be merged in